### ISSUE TITLE
Crash when using provided code snippet to redirect NSLog

### ISSUE THREAD

**Issue by allenhumphreys** [Instabug/Instabug-iOS#144]
SDK version 7.1.0

As provided by your [documentation](http://docs.instabug.com/docs/logging)
```
inline void NSLog(NSString *format, ...) {
    va_list arg_list;
    va_start(arg_list, format);
    IBGNSLog(format, arg_list);
    va_end(arg_list);
}
```

Getting a lot of these crashes:

<img width="1473" alt="screen shot 2017-03-25 at 1 39 19 pm" src="https://cloud.githubusercontent.com/assets/8225090/24325134/eaa68976-1160-11e7-97de-c3170f9bd673.png">

**Comment by Kmohamed** [comment]
Let me check it and get back to you.

**Comment by HeshamMegid** [comment]
@allenhumphreys we're seeing hundreds of logs with the same message in that crash. Is that intentional? If it is, then we would appreciate if you create a sample project that has this logging so we could try to reproduce the crash on our side.

**Comment by allenhumphreys** [comment]
@HeshamMegid I don't think that is intentional and have never seen that happen before during debugging. Is it possible that the log is somehow recursing unintentionally?

**Comment by allenhumphreys** [comment]
@HeshamMegid Followup, the repeated log messages have nothing to do with it.

In an AppDelegate, you can do this:
```
<Start Instabug>

AVURLAsset *asset = [AVURLAsset assetWithURL:[NSURL fileURLWithPath:@"/var/mobile/Containers/Data/Application/8040F1BC-66AA-41AD-BD53-3A733B1CE354/Documents/2017 Syracuse Stadium Cross Pit Party.mp4"]]; // The file doesn't need to exist for this to work

NSLog(@"%@", asset);
```

It will crash with a pretty high reliability. The issue seems to be related to possibly out-of-bounds memory access or improper string handling.

<img width="307" alt="screen shot 2017-03-26 at 3 35 21 pm" src="https://cloud.githubusercontent.com/assets/8225090/24334960/e12b7fda-1239-11e7-99cc-1cae99d14641.png">

**Comment by Kmohamed** [comment]
@allenhumphreys Thanks a lot for sending this sample.
It really helps to us to reproduce the issue it will be fixed in next SDK version v7.1.1
Also I can send you a custom builds if you want.
